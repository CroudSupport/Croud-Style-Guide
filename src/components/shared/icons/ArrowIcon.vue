
<script>
    /**
    * Standard arrow icon to use in Croud Control
    *
    * *This is a functional component which has no state so does not need to define any props*
    * @see https://vuejs.org/v2/guide/render-function.html#Functional-Components
    *
    * @example ./croud-arrow-icon.md
    */
    export default {
        functional: true,

        name: 'croud-arrow-icon',

        render(h, context) {
            console.log(context)
            return h('i', {
                class: {
                    chevron: true,
                    icon: true,
                    [context.props.direction || 'right']: true,
                    ...context.data.class || {},
                },
            })
        },
    }
</script>

