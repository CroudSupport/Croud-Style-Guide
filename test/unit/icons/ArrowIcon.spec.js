
import Vue from 'vue'
import ArrowIcon from '../../../src/components/shared/icons/ArrowIcon'

const classes = {}
const props = {}

const Constructor = Vue.extend({
    render(h) {
        return h(ArrowIcon, {
            props,
            class: classes,
        })
    },
})

const vm = new Constructor().$mount()

describe('ArrowIcon', () => {
    it('should match the snapshot', () => {
        expect(vm.$el).toMatchSnapshot()
    })

    it('should change direction', () => {
        props.direction = 'left'
        vm.$forceUpdate()
        vm.$nextTick(() => {
            expect(vm.$el).toMatchSnapshot()
        })
    })

    it('should accept classes', () => {
        classes.disabled = true
        vm.$forceUpdate()
        vm.$nextTick(() => {
            expect(vm.$el).toMatchSnapshot()
        })
    })
})
