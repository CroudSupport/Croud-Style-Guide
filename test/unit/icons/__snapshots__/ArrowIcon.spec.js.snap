// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowIcon should accept classes 1`] = `
<i
  class="chevron icon left disabled"
/>
`;

exports[`ArrowIcon should change direction 1`] = `
<i
  class="chevron icon left"
/>
`;

exports[`ArrowIcon should match the snapshot 1`] = `
<i
  class="chevron icon right"
/>
`;
